src/renderer/components/InvestTab/index.jsx

```diff
@@ -110,58 +110,66 @@ export default class InvestTab extends React.Component {
     // in part by it's workspace directory.
     args.workspace_dir = path.resolve(argsValues.workspace_dir);
 
-    job.setProperty('argsValues', args);
-    job.setProperty('status', 'running');
-
-    // Setting this very early in the click handler so the Execute button
+    // Setting status very early in the click handler so the Execute button
     // can display an appropriate visual cue when it's clicked
-    this.setState({
-      jobStatus: job.metadata.status,
-      workspaceDir: args.workspace_dir
+    updateJobProperties(jobID, {
+      argsValues: args,
+      status: 'running',
     });
-    ipcRenderer.on(`invest-logging-${this.ipcSuffix}`, (event, logfile) => {
-      this.setState({
-        logfile: logfile
-      }, () => {
-        this.switchTabs('log');
-      });
-      job.setProperty('logfile', logfile);
-      saveJob(job);
+
+    ipcRenderer.on(`invest-logging-${jobID}`, (event, logfile) => {
+      updateJobProperties(jobID, { logfile: logfile });
+      this.switchTabs('log');
     });
-    ipcRenderer.on(`invest-stderr-${this.ipcSuffix}`, (event, data) => {
+    ipcRenderer.on(`invest-stderr-${jobID}`, (event, data) => {
       let stderr = Object.assign('', this.state.logStdErr);
       stderr += data;
       this.setState({
         logStdErr: stderr,
       });
     });
-    ipcRenderer.on(`invest-exit-${this.ipcSuffix}`, (event, code) => {
+    ipcRenderer.on(`invest-exit-${jobID}`, (event, code) => {
       // Invest CLI exits w/ code 1 when it catches errors,
       // Models exit w/ code 255 (on all OS?) when errors raise from execute()
       // Windows taskkill yields exit code 1
       // Non-windows process.kill yields exit code null
       const status = (code === 0) ? 'success' : 'error';
-      this.setState({
-        jobStatus: status
+      let finalTraceback = '';
+      if (this.state.logStdErr) {
+        let i = 1;
+        while (!finalTraceback) {
+          [finalTraceback] = `${this.state.logStdErr}`

```

Comment:
Just trying to walk through this while loop. If we have some stderr we'll parse it by line until `finalTraceback` will eventually be empty? what exactly is the `-1 * i` splice doing?

Reply:
Whew, that's a doozy of a question. This definitely deserves a code-comment. The idea here is to take the final line of stdErr and set it aside as "finalTraceback" for use in an `Alert`. The problem (I guess...it's been a while) is sometimes there are blank lines at the end of the stderr. So `splice(-1)` takes the last line in the array, but then keeps moving up from the back of the array by `i` until it finds a not blank line.

Reply:
I think this can probably become a whole lot more readable using `pop`. And in the process I'm making sure this is covered by a test. More soon.